net/golioth/settings.c

```diff
@@ -95,6 +96,27 @@ static int send_status_report(struct golioth_client *client,
 	return send_coap_response(client, response_buf, nbytes);
 }
 
+static void save_setting(const char *key, const struct golioth_settings_value *value)
+{
+	int err;
+	char name[64] = {};
+
+	/*
+	 * Use the "gsettings" package in the Zephyr settings subsystem
+	 * to store all settings.
+	 */
+	snprintf(name, sizeof(name), "gsettings/%s", key);
+
+	if (IS_ENABLED(CONFIG_GOLIOTH_SETTINGS_PERSIST)) {
+		LOG_INF("Saving setting: %s", name);
+		err = settings_save_one(name, value, sizeof(struct golioth_settings_value));
```

Comment:
Nit: I prefer to use `sizeof(*value)`, so if the type changes in function argument, we don't have to update function body. It is also easier to verify that proper size is used (it is all part of single line, no need to look into the whole function to check if `value` is really a pointer to `struct golioth_settings_value`).

What do you think?